streams/src/main/java/org/apache/kafka/streams/state/internals/ForwardingStateStoreSupplier.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
```

Comment:
Two further things: 
1. the cache is a `byte[]` cache, so if this was exposed the `Processor` would also need to have access to the Serdes.
2. you can't just call `context.forward()` when a flush/eviction happens as it could happen at any stage, i.e., before or after `this` node in the topology. You'd need to set the `ProcessorRecordContext` before calling `forward`